### ISSUE TITLE
JS Integration tests

### ISSUE THREAD

**Issue by wilwade** [LibertyDSNP/frequency#371]
JS Integration tests would be a great way to both have examples and do full tests from top to bottom. Cross extension to RPC calls to testing the api-augmentation.

## Examples
- https://github.com/AcalaNetwork/Acala/tree/master/ts-tests

## Notes
- @demisx to help with the CI portion
- mocha is suggested as the most common for web3 uses
- May need to setup a manual sealing version

**Comment by rlaferla** [comment]
This may be useful:  https://joelhooks.com/jest-and-github-actions

**Comment by rlaferla** [comment]
What folder/file structure would work best?  Some tests will be straightforward but others may require more than one script file and data files.
```
src/tests 
          ├── test1
          │       ├── func.js
          │       ├── func.test.js
          │       └── .....
          ├── test2  
          │       ├── func.js
          │       ├── func.test.js
          │       ├── data.json
          │       └── .....
          ├── ....
```

**Comment by dopatraman** [comment]
do we have an idea about what workflows we should be testing?

**Comment by wilwade** [comment]
> do we have an idea about what workflows we should be testing?

Now that we are more stable we could have a deployment integration test. I think that is the only integration test we would want.

Wouldn't be too hard to setup with the docker container running as a service in GitHub Actions.

**Comment by wilwade** [comment]
> do we have an idea about what workflows we should be testing?

I think most of them will be basic happy path. Create a schema, create a message, create an msa.

Perhaps also some replay attack e2e tests

**Comment by wilwade** [comment]
Closed with #637
